bowtie/_report.py

```diff
@@ -9,6 +10,7 @@
 from attrs import asdict, field, frozen
 from attrs.filters import exclude
 from rpds import HashTrieMap
+from semver import VersionInfo
```

Comment:
An arbitrary implementation is not really guaranteed to be using semver (or any semver-resembling scheme) -- so we anyways would need to potentially handle arbitrary version schemes, which makes me think unlike in general (where we certainly should use rich types whenever possible) it's not really worthwhile to use some package with a specific versioning format -- e.g. what will this do for an implementation with version `20+foo`? Will it blow up? If so it seems like we should likely just roll our own simple thing which splits on `.` + `int` but otherwise falls back to simple lexicographic, and doesn't really do anything else (any other validation), unless this package or some other really does do exactly that.

Reply:
So in that case, before I start writing the custom thing, should we consider `4.18.0a10` to be greater than `4.18.0` ? and also `20+foo` to be greater than `20` ?

Reply:
I think it's probably fine for those to sort lexicographically, we likely won't really test prerelease versions anyhow.